internal/browser/seperator.go

```diff
@@ -0,0 +1,58 @@
+package browser
+
+import (
+	"fmt"
+	"strings"
+	"sync/atomic"
+
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+// progressState is the global flag to show/hide progress
+var progressState int32
+
+const (
+	HideProgress = iota
+	ShowProgress
+)
+
+var spinnerWidget = spinner.New()
+
+func changeProgress(ps int32) {
+	switch {
+	case ps == HideProgress:
+		if atomic.CompareAndSwapInt32(&progressState, ShowProgress, HideProgress) {
+			return
+		}
+	case ps == ShowProgress:
+		if atomic.CompareAndSwapInt32(&progressState, HideProgress, ShowProgress) {
+			return
+		}
+	}
+}
+
+type SeparatorWithProgress struct {
+	length int
+}
+
+func (s *SeparatorWithProgress) Init() tea.Cmd {
+	return spinnerWidget.Tick
+}
+
+func (s *SeparatorWithProgress) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	if msg, ok := msg.(tea.WindowSizeMsg); ok {
+		s.length = msg.Width - 2
```

Comment:
What is `-2`? Does it make sense to move it to a named constant?

Reply:
Removed it completely https://github.com/hazelcast/hazelcast-commandline-client/pull/104/commits/925164c7c36af50ed95e949a2686edda2bce66f2